doc/classes/Object.xml

```diff
@@ -22,6 +22,7 @@
 		Lastly, every object can also contain metadata (data about data). [method set_meta] can be useful to store information that the object itself does not depend on. To keep your code clean, making excessive use of metadata is discouraged.
 		[b]Note:[/b] Unlike references to a [RefCounted], references to an object stored in a variable can become invalid without being set to [code]null[/code]. To check if an object has been deleted, do [i]not[/i] compare it against [code]null[/code]. Instead, use [method @GlobalScope.is_instance_valid]. It's also recommended to inherit from [RefCounted] for classes storing data instead of [Object].
 		[b]Note:[/b] The [code]script[/code] is not exposed like most properties. To set or get an object's [Script] in code, use [method set_script] and [method get_script], respectively.
+		[b]Note:[/b] In a boolean context, an [Object] will evaluate to [code]false[/code] if it is equal to [code]null[/code] or it has been freed. Otherwise, an [Object] will always evaluate to [code]true[/code]. See also [method @GlobalScope.is_instance_valid].
```

Comment:
```suggestion
		[b]Note:[/b] In a boolean context, an [Object] will evaluate to [code]false[/code] if it is equal to [code]null[/code] or it is an invalid object (e.g. it has been deleted from memory). Otherwise, an [Object] will always evaluate to [code]true[/code]. See also [method @GlobalScope.is_instance_valid].
```

Since there is `Object#null` and for symmetry with `is_instance_valid()` docs:

https://github.com/godotengine/godot/blob/ec6a1c0e792ac8be44990749800a4654a293b9ee/doc/classes/%40GlobalScope.xml#L575

Also, it's worth adding a note to the `is_instance_valid()` docs that since 4.4 this function is mostly optional in GDScript, you get the same result whether you explicitly or implicitly cast an object to `bool`.

Reply:
Once more there's an ambiguous "e.g." here. Just for clarity, it means less "That would mean" and more "Here's one example of invalid object"? Either way, I may come up with another way to word it.

Reply:
> Just for clarity, it means less "That would mean" and more "Here's one example of invalid object"? I assume the latter.

Yes, "invalid object" = `Object#null` | `Freed Object`.